### PR TITLE
Fix for 1.31.1

### PR DESCRIPTION
**Pull request by aaronjamt**
These changes allow the project to compile successfully for 1.31.1 and actually do work in game

### CODE REVIEW

File: Counters+/UI/MenuButtonManager.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ private void SummonFlowCoordinator()
 
         public void Dispose()
         {
-            if (MenuButtons.IsSingletonAvailable && BSMLParser.IsSingletonAvailable)
```
Comment (aaronjamt) [inline Counters+/UI/MenuButtonManager.cs:33]: This part might be wrong, the original IsSingletonAvailable checked internal private variables that are not really possible to check anymore. I feel like this works "well enough" but may need to be changed around or use something else instead

Comment (aaronjamt) [reply Counters+/UI/MenuButtonManager.cs:33]: > BSML got updated to ensure all the singleton classes were always available, so you're fine removing this check for them

Ah, okay. That explains why I wasn't able to find any examples of how other projects checked for them. I've updated the PR.

**Comment by aaronjamt** [comment]
Just converted this to a draft, turns out that it works fine in normal gameplay but the settings menu is broken.

**Comment by aaronjamt** [comment]
It's something to do with the BSML `CustomListTableData`, specifically the `.data` property. For some reason, both the getter and setter seem to just... not exist? BSML throws an error when trying to access the property, saying that it is trying to Invoke `.get_data()` or `.set_data()` via reflection.

**Comment by aaronjamt** [comment]
Fixed that, we're good for a merge now (and I now think the `CustomListTableData` think was an unrelated issue specific to my machine)

File: Counters+/Utils/SharedCoroutineStarter.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
+using UnityEngine;
+
+namespace CountersPlus.Utils
+{
+    // Mostly a copy-paste of PersistentSingleton<T> from game version < 1.31.0.
+    internal class SharedCoroutineStarter : MonoBehaviour
```
Comment (Caeden117) [inline Counters+/Utils/SharedCoroutineStarter.cs:6]: Is there not a re-implementation of this class in another library? I'm curious if a Counters+ specific implementation is necessary.

File: Counters+/Utils/SharedCoroutineStarter.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                {
+                    if (_instance == null)
+                    {
+                        _instance = FindObjectOfType<SharedCoroutineStarter>();
+
+                        if (FindObjectsOfType<SharedCoroutineStarter>().Length > 1)
+                        {
+                            Debug.LogError("[Singleton] Something went really wrong - there should never be more than 1 singleton! Reopening the scene might fix it.");
+                            return _instance;
+                        }
```
Comment (Caeden117) [inline Counters+/Utils/SharedCoroutineStarter.cs:32]: Not sure if this code is necessary - `_instance` should never be null when a `SharedCoroutineStarter` object already exists.

If you want some extra assurance, I'd probably move the `_instance` assignment to the `OnEnable` method of this component.

File: Counters+/Utils/SharedCoroutineStarter.cs
[the remark this answers is not in the document]
Comment (Caeden117) [reply Counters+/Utils/SharedCoroutineStarter.cs:1]: Trust me, if I want a better way of doing anything, I'd just completely rewrite Counters+ from scratch. At this point, a little more jank won't hurt.

File: Counters+/Utils/SharedCoroutineStarter.cs (hunk 3 above)
Comment (aaronjamt) [reply Counters+/Utils/SharedCoroutineStarter.cs:32]: This method is copy+pasted from the old version (before they removed it from the base game). Will look for another library or something to replace it

File: Counters+/Utils/SharedCoroutineStarter.cs (hunk 2 above)
Comment (aaronjamt) [reply Counters+/Utils/SharedCoroutineStarter.cs:6]: This method is copy+pasted from the old version (before they removed it from the base game). Will look for another library or something to replace it

File: Counters+/Utils/SharedCoroutineStarter.cs
[the remark this answers is not in the document]
Comment (aaronjamt) [reply Counters+/Utils/SharedCoroutineStarter.cs:1]: Haha, fair enough

**Pull request merged**